### ISSUE TITLE
UnicodeError due to non-ASCII transaction notes

### ISSUE THREAD

**Issue by mgedmin** [mgedmin/zodbbrowser#12]
Traceback (mostly useless):

```
Traceback (most recent call last):
  File "/srv/zopes/akl-2.13/eggs/zope.publisher-3.12.6-py2.6.egg/zope/publisher/publish.py", line 132, in publish
    result = publication.callObject(request, obj)
  File "/srv/zopes/akl-2.13/eggs/zope.app.publication-3.12.0-py2.6.egg/zope/app/publication/zopepublication.py", line 207, in callObject
    return mapply(ob, request.getPositionalArguments(), request)
  File "/srv/zopes/akl-2.13/eggs/zope.publisher-3.12.6-py2.6.egg/zope/publisher/publish.py", line 107, in mapply
    return debug_call(obj, args)
   - __traceback_info__: <bound method ZodbInfoView.__call__ of <zope.browserpage.metaconfigure.ZodbInfoView object at 0xb72ca8c>>
  File "/srv/zopes/akl-2.13/eggs/zope.publisher-3.12.6-py2.6.egg/zope/publisher/publish.py", line 113, in debug_call
    return obj(*args)
  File "/srv/zopes/akl-2.13/eggs/zodbbrowser-0.12.0-py2.6.egg/zodbbrowser/browser.py", line 94, in __call__
    return self.render()
  File "/srv/zopes/akl-2.13/eggs/zodbbrowser-0.12.0-py2.6.egg/zodbbrowser/browser.py", line 153, in render
    return self.template()
  File "/srv/zopes/akl-2.13/eggs/zope.browserpage-3.12.2-py2.6.egg/zope/browserpage/viewpagetemplatefile.py", line 83, in __call__
    return self.im_func(im_self, *args, **kw)
  File "/srv/zopes/akl-2.13/eggs/zope.browserpage-3.12.2-py2.6.egg/zope/browserpage/viewpagetemplatefile.py", line 51, in __call__
    sourceAnnotations=getattr(debug_flags, 'sourceAnnotations', 0),
  File "/srv/zopes/akl-2.13/eggs/zope.pagetemplate-3.5.2-py2.6.egg/zope/pagetemplate/pagetemplate.py", line 114, in pt_render
    return output.getvalue()
  File "/usr/lib/python2.6/StringIO.py", line 270, in getvalue
    self.buf += ''.join(self.buflist)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc5 in position 30: ordinal not in range(128)
```

**Comment by mgedmin** [comment]
A patch that helps debugging:
```diff
--- /srv/zopes/akl-2.13/eggs/zodbbrowser-0.12.0-py2.6.egg/zodbbrowser/browser.py.orig	2015-02-27 16:40:42.000000000 +0200
+++ /srv/zopes/akl-2.13/eggs/zodbbrowser-0.12.0-py2.6.egg/zodbbrowser/browser.py	2015-02-27 16:35:59.000000000 +0200
@@ -92,6 +92,8 @@
     def __call__(self):
         try:
             return self.render()
+        except:
+            import pdb; pdb.post_mortem()
         finally:
             if self.readonly or not self.made_changes:
                 resources = transaction.get()._resources

```

**Comment by mgedmin** [comment]
pdb session:
```
> /usr/lib/python2.6/StringIO.py(270)getvalue()
-> self.buf += ''.join(self.buflist)
(Pdb) pp filter(lambda x: type(x) is str, self.buflist)
['PersistentOutline',
 '/.../broken object',
 "&lt;class 'Products.ZWiki.OutlineSupport.PersistentOutline'&gt;",
 '_childmap',
 '_nesting',
 '_parentmap',
 'mantas',
 '/akl/wiki/manage_renameObjects',
 'mantas',
 '/akl/wiki/manage_pasteObjects',
 'mantas',
 '/akl/wiki/FrontPage/edit\n\n"Suk\xc5\xabriau pradin\xc4\xaf Pirtis2012 pakvietim\xc4\x85 pagal Mykolo OK naujien\xc4\x85"',
 'mantas',
 '/akl/wiki/FrontPage/edit\n\n"Info apie ZiemosMytas2011 - susimatym\xc4\x85 Mati\xc5\xa1k\xc4\x97s kaimo sodybos pirtel\xc4\x97je, 2011 sausio 22d."',
 'grazulis',
 '/akl/wiki/FrontPage/edit',
 '1.628s |',
 '0.12.0']
```
If I do `pp self.buflist` without the filtering, I see that the non-ascii bits are wrapped in `<span class="description">`:
```
(Pdb) pp self.buflist
...
 u'<div class="transaction">',
 u'\n        ',
 u'<h4 class="transaction" id="tid255489849326903449">',
 u'\n          ',
 u'<a class="subtitle"\n             href="@@zodbbrowser_history?tid=255489849326903449">',
 u'view transaction record</a>\n          ',
 u'<a class="title"\n             href="@@zodbbrowser?oid=0x7d3b3&amp;tid=0x38baeaa10d92099">',
 u'\n            #',
 u'2',
 u':\n            ',
 u'<span class="timestamp" title="UTC">',
 u'2011-01-18 15:06:03',
 u'</span>\n            ',
 u'<span class="user" title="user from site akl">',
 'mantas',
 u'</span>\n            ',
 u'<span class="location" title="request type None">',
 u'</span>\n            ',
 u'<span class="description">',
 '/akl/wiki/FrontPage/edit\n\n"Info apie ZiemosMytas2011 - susimatym\xc4\x85 Mati\xc5\xa1k\xc4\x97s kaimo sodybos pirtel\xc4\x97je, 2011 sausio 22d."',
 u'</span>\n          </a>\n        </h4>\n        ',
...
```

**Comment by mgedmin** [comment]
Proposed fix: https://gist.github.com/mgedmin/391acd3828424e856d0c

**Comment by mgedmin** [comment]
I'm pretty sure this was fixed in d9eb09b77ee72213b75a5001575a6085951f480d, during the Python 3 port.
